pkg/image/build.go

```diff
@@ -156,6 +167,38 @@ func Build(cfg *config.Config, dir, imageName string, secrets []string, noCache,
 		global.LabelNamespace + "has_init": "true",
 	}
 
+	if isUsingCogBaseImage {
+		labels[global.LabelNamespace + "using_cog_base_image"] = "true"
+		labels[global.LabelNamespace + "cog-base-image-name"] = cogBaseImageName
+
+		// get the last layer of the cog base image so that when we look at the built cog image,
+		// we know where the base image ends
+		// pull the base image, as we'll need to pull it anyway to build the cog image
+		// Q: are we ok with the cog base image being pulled to the user's machine? can fix this later by implementing 
+		// a manifest inspect which doesn't require a pull
+
+		err := docker.Pull(cogBaseImageName)
+		if err != nil {
+			return fmt.Errorf("Failed to pull cog base image: %w", err)
+		}
+
+		cogBaseImage, err := docker.ImageInspect(cogBaseImageName)
+		if err != nil {
+			return fmt.Errorf("Failed to inspect cog base image while trying to fetch last layer: %w", err)
+		}
+
+		if cogBaseImage.RootFS.Layers == nil || len(cogBaseImage.RootFS.Layers) == 0 {
+			return fmt.Errorf("Cog base image has no layers or RootFS is nil: %s", cogBaseImageName)
+		}
+
+		lastLayerIndex := len(cogBaseImage.RootFS.Layers) - 1
+		lastLayer := cogBaseImage.RootFS.Layers[lastLayerIndex]
+		console.Debugf("Last layer of the cog base image: %s", lastLayer) // prints the sha
+
+		labels[global.LabelNamespace + "cog-base-image-last-layer-diff-id"] = lastLayer
+		labels[global.LabelNamespace + "cog-base-image-last-layer-idx"] = fmt.Sprintf("%d", lastLayerIndex)
```

Comment:
Do we need this information if we have the layer identifier itself?

```suggestion
```

Reply:
Yes - this is an optimization for the image conversion pipeline so that rather than looking through the shas until we find a match, we can directly start looking for non-base layers starting at cog-base-image-last-layer-idx + 1. We double check things - we check the layer at the index, and check that its value matches the expected sha, in the pipeline, just in case the image somehow got tampered with or an extra layer was added.